src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs

```diff
@@ -23,6 +25,45 @@ public HttpResponse GetResponse(HttpRequest request, CookieContainer cookies)
             webRequest.ContentLength = 0;
             webRequest.CookieContainer = cookies;
 
+            if (request.Proxy != null)
+            {
+                IPAddress[] addresses = Dns.GetHostAddresses(request.Proxy.Host);
+                switch (request.Proxy.Type)
+                {
+                    case ProxyType.Http:
+                        if(request.Proxy.Username != null && request.Proxy.Password != null)
```

Comment:
please extract `request.Proxy.Username != null && request.Proxy.Password != null` into a local varible something like `authenticated` to remove the duplicated checks.

also check for null as well as empty strings.